conjure-core/src/main/java/com/palantir/conjure/defs/validator/EndpointDefinitionValidator.java

```diff
@@ -136,14 +138,35 @@ public void validate(EndpointDefinition definition) {
                         .stream()
                         .anyMatch(entry -> entry.getParamType().accept(ParameterTypeVisitor.IS_BODY));
 
-                Preconditions.checkState(!hasBody,
+                Preconditions.checkState(
+                        !hasBody,
                         "Endpoint cannot be a GET and contain a body: method: %s, path: %s",
                         definition.getHttpMethod(),
                         definition.getHttpPath());
             }
         }
     }
 
+    @com.google.errorprone.annotations.Immutable
+    private static final class NoOptionalBinaryBodyParamValidator implements ConjureValidator<EndpointDefinition> {
+        @Override
+        public void validate(EndpointDefinition definition) {
+            boolean hasOptionalBinaryBody = definition.getArgs()
+                    .stream()
+                    .filter(entry -> entry.getParamType().accept(ParameterTypeVisitor.IS_BODY))
+                    .map(ArgumentDefinition::getType)
+                    .filter(type -> type.accept(TypeVisitor.IS_OPTIONAL))
```

Comment:
You'll want to use a dealiasingTypeVisitor, like we do in other validators, to protect against alias of `optional<binary>`

Reply:
like in NoComplexPathParamValidator, ja?

Reply:
done